### DISCUSSION TITLE
Support C2PA manifests preservation

### DISCUSSION THREAD

**Discussion by HalfdanJ** [imgproxy/imgproxy#1329] [Ideas]
In our company we are looking into start supporting the [C2PA initiative](https://c2pa.org/), to add manifest information that allows tracking the source of the image (specifically for us to trace generative media). Our biggest obstacle is how to maintain this manifest as the images get served to our clients with imgproxy, since it seems like imgproxy always will strip this away even when serving the same file format as the original. 

I know way too little about the technical details of both c2pa and imgproxy to be able to determine how feasible or infeasible it even is to support this, but wanted to bring this up for discussion since it might become an increasing issue loosing this traceability when serving images. 

The c2pa standard has taken time to take off, but for example photoshop supports it now, and most of the major generative media generators do. And just now, google has announced they will also use it to [tag content in google search](https://www.theverge.com/2024/9/17/24247004/google-c2pa-verify-ai-generated-images-content)

**Comment by DarthSim** [comment]
Hey @HalfdanJ!

This is a good idea, but its implementation may require much time.

First of all, the only mature C2PA SDKs at the moment are Rust and JS.  JS one is a client library, and we can not use JS in imgproxy, leaving us only with Rust SDK. We'll need to write some kind of wrapper around it.

Next, though Rust SDK allows reading raw JUMBF metadata from files and writing it back, I'm not sure if this will work and if this is a good idea. I am not familiar with C2PA at all, so this needs to be investigated.

**Comment by HalfdanJ** [comment]
Thanks for the response, i did fear it would be a complicated feature. 

One thing i did realize after posting is that i can disable processing of certain file formats using `IMGPROXY_SKIP_PROCESSING_FORMATS`. This helps us somewhat in that we can preserve the manifest when serving the original files, and we only loose it when making thumbnails.

**Comment by DarthSim** [comment]
You also can use the `raw` processing option to stream files as-is.
